Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -84,6 +84,38 @@ open class ChatMessageListVC: _ViewController,
         return (!listView.isLastCellFullyVisible && isMoreContentThanOnePage) || dataSource?.isFirstPageLoaded == false
     }
 
+    /// A button to scroll the collection view to the first unread message.
+    /// Visible when there are unread messages outside of the bounds of the screen.
+    open private(set) lazy var jumpToUnreadMessagesButton: JumpToUnreadMessagesButton = components
+        .jumpToUnreadMessagesButton
+        .init()
+        .withoutAutoresizingMaskConstraints
+
+    /// A Boolean value indicating whether jump to unread messages button is visible.
+    open var showJumpToUnreadMessages: Bool {
+        guard let dataSource = dataSource,
+              dataSource.sessionUnreadCount.messages > 0 else {
+            return false
+        }
+
+        guard let lastReadIndexPath = lastReadMessageIndexPath else {
+            // If there are messages, but none is read, we show the button if the first message is not on screen.
```

Comment:
Wouldn't the messages that are onscreen be marked as read?

Reply:
We mark the channel as read only when all the unread messages are seen